### ISSUE TITLE
l3color: add a test if a name color has been declared. 

### ISSUE THREAD

**Issue by u-fischer** [latex3/latex3#708]
Currently this simply errors:

~~~~
\documentclass{article}
\usepackage{l3color}
\ExplSyntaxOn
\color_parse:nN {myblue}\l_tmpa_tl
\ExplSyntaxOff
\begin{document}
some text
\end{document}
~~~~

It would be nice if one could test first if the color exists (without checking the internal variable). It would be even nicer if they were a way to check complex expressions like `myblue!10!red` to. The background is that currently it is not really possible to decide which package (color, xcolor or l3color) defined a color and that makes it a bit difficult to get the color values.

**Comment by blefloch** [comment]
This seems to call for a

\color_parse:nNTF {myblue} \l_tmpa_tl
  { The color expression is valid and stored in \l_tmpa_tl }
  { Invalid and \l_tmpa_tl is set to \q_no_value? or perhaps to the
first argument? }

**Comment by josephwright** [comment]
Testing if a named colour is defined is relatively easy: we just have to look for

- `\csname \string \color @<name>\endcnsame`
- ` \csname l__color_named_<name>_tl\endcsname`

and can then extract the model, etc., pretty easily. If we want to deal with full expressions, life gets more 'interesting'. One method would be for example to allow `\color_parse:nN` to examine both 'native' colours and those from `color`/`xcolor`, such that an expression will parse if it's valid _at all_. That would require some additional code to grab-and-convert 'non-native' colours. Another plan, as Bruno says, would be to have a `TF` version that simply does nothing for an invalid colour. But that then needs parsing by `xcolor` too to reliably extract the info. Sounds more awkward than extending `l3color` so it can look up `color`/`xcolor` data: thoughts?

**Comment by josephwright** [comment]
@u-fischer  If `\color_parse:nN` could also read `color`/`xcolor` colours, would that help? Is the output of `\color_parse:nN` suitable? Also, do you need the 'full' `xcolor` expressions or are the 'simple' ones enough: currently `l3color` only does the latter.

**Comment by josephwright** [comment]
Fixed by db2b264d50e658fa893b9d032a436ed33639ca15
